README.md

```diff
@@ -1,8 +1,362 @@
 # CUDA C++ Core Libraries (CCCL)
 
-The future home of the CUDA C++ Core Libraries (Thrust, CUB, libcu++).
+Welcome to the CUDA C++ Core Libraries (CCCL) where our mission is to make CUDA C++ more delightful.
+
+This repository unifies three essential CUDA C++ libraries into a single, convenient repository: 
+
+1. [Thrust](https://github.com/nvidia/thrust)
+2. [libcudacxx](https://github.com/nvidia/libcudacxx)
+3. [CUB](https://github.com/nvidia/cub)
+
+Our goal is to provide CUDA C++ developers with the building blocks they need to make it easier to write safe and efficient code. By bringing these libraries together, we hope to streamline your development process and broaden your ability to leverage the power of CUDA C++.
+For more information about our decision to unify these projects, see our announcement here: (TODO)
+
+## Overview
+
+The concept for the CUDA C++ Core Libraries (CCCL) grew organically out of the Thrust, CUB, and libcudacxx projects that were developed independently over the years with a similar goal: to provide a high-quality, high-performance, and easy-to-use C++ abstractions for CUDA developers.
+Naturally, there was a lot of overlap among the three projects, and we realized that we could better serve the community by unifying them into a single repository.
+
+- **Thrust** is the C++ parallel algorithms library which inspired the introduction of parallel algorithms to the C++ Standard Library. Thrust's high-level interface greatly enhances programmer productivity while enabling performance portability between GPUs and multicore CPUs via configurable backends that allow using multiple parallel programming frameworks (such as CUDA, TBB, and OpenMP).
+
+- **libcudacxx** is the CUDA C++ Standard Library. It provides an implementation of the C++ Standard Library that works in both host and device code. Additionally, it provides abstractions for CUDA-specific hardware features like synchronization primitives, cache control, atomics, and more. 
+
+- **CUB** is a lower-level, CUDA-specific library designed for speed-of-light parallel algorithms across all GPU architectures. In addition to device-wide algorithms, it provides *cooperative algorithms* like block-wide reduce and warp-wide scan, providing CUDA kernel developers with building blocks to create speed-of-light, custom kernels. 
+
+Bringing these libraries together enables us to better deliver the features you need as well as provide a more cohesive experience where everything “just works” out of the box.  
+Our long-term goal is to offer a single, unified, modern C++ library that merges and extends the functionality of our existing libraries and more. 
+We envision the CUDA C++ Core Libraries to fill a similar role that the Standard C++ Library fills for Standard C++ – equipping you with general-purpose, speed-of-light tools to focus on solving the problems you care about, not boilerplate.
```

Comment:
I'm not sure I follow the "not boilerplate" part

Reply:
Yeah, for people who didn't attend Jake's internal talk, the "boilerplate" sentence seems to come up without proper context.